mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/LinkBankAccountScreen.kt

```diff
@@ -0,0 +1,324 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Divider
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import org.mifos.mobilewallet.mifospay.R
+import org.mifos.mobilewallet.mifospay.bank.presenter.BankUiState
+import org.mifos.mobilewallet.mifospay.bank.presenter.LinkBankAccountViewModel
+import org.mifos.mobilewallet.mifospay.designsystem.component.MfOverlayLoadingWheel
+import org.mifos.mobilewallet.mifospay.designsystem.component.MifosOutlinedTextField
+import org.mifos.mobilewallet.mifospay.designsystem.component.MifosTopBar
+import org.mifos.mobilewallet.mifospay.designsystem.theme.greyBackgroundColor
+import org.mifos.mobilewallet.mifospay.domain.model.Bank
+import org.mifos.mobilewallet.mifospay.theme.MifosTheme
+
+@Composable
+fun LinkBankAccountScreen(
+    viewModel: LinkBankAccountViewModel = hiltViewModel(),
+    context: Context = LocalContext.current,
+) {
+
+    var showSimBottomSheet by rememberSaveable { mutableStateOf(false) }
+    var bankSelected = ""
+
+
+    Scaffold(topBar = {
+        MifosTopBar(
+            topBarTitle = R.string.link_bank_account,
+            backPress = {
+                (context as Activity).onBackPressed()
+            }
+        )
+    }) {
+
+        LaunchedEffect(key1 = true) {
+            viewModel.getAllBanks(context)
+
+        }
+
+        val uiState by viewModel.bankState.collectAsState()
+
+
+        LinkBankAccountScreenContent(it, uiState, viewModel) {
+            bankSelected = it.name
+            showSimBottomSheet = true
+        }
+
+        if (showSimBottomSheet) {
```

Comment:
pass this as field in ChooseSimDialogSheet  and then use

Reply:
```
ChooseSimDialogSheet(
       showBottomSheet: Boolean ,
      ... // other parameters
)
``` 
I thought we can implement something like this but if thats the case then I think we can go ahead with this